### ISSUE TITLE
DateFormat.getBestDateTimePattern() does not return the correct format

### ISSUE THREAD

**Issue by aadithyb** [robolectric/robolectric#2665]
I am using getBestDateTimePattern() available in Android's `DateFormat` class to do some date formatting. It runs perfectly when I run the app but does not when I try to run some tests on it. 

getBestDateTimePattern(Locale.getDefault(), "EEEMMMd") while running a unit test does not create the necessary formatting which is `EEE, MMM, d` which happens only on the device. 

Robolectric version: 3.1.2

Possibly related to https://github.com/robolectric/robolectric/pull/2513

**Comment by JSchaenzle** [comment]
Yeah, I experienced the same thing. The problem is that this method just isn't implemented in the ShadowDateFormat class that robolectric provides. Implementing it in the shadow might be outside the scope of this project. I just ended up testing this functionality in an integration test instead of a unit test - that way I could use the native android lib.

On this note, it would be nice if robolectric stubbed out these methods with an exception or something so that it was more obvious why it fails rather than just the no-method exception...
